src/stores/filingHistoryListStore.ts

```diff
@@ -0,0 +1,322 @@
+            await Promise.all(promises)
+
+            // leave busy spinner displayed another 250ms
+            // (to mitigate flashing when the promises are resolved quickly)
+            setTimeout(() => { rootStore.setFetchingDataSpinner(false) }, 250)
+          }
+        }
+
+        this.setPanel(isCurrentPanel ? null : index)
+        resolve()
+      })
+    },
+
+    /** Loads the comments for this history item. */
+    async loadComments (filing: ApiFilingIF): Promise<void> {
+      try {
+      // fetch comments array from API
+        const comments = await LegalServices.fetchComments(filing.commentsLink)
+        // flatten and sort the comments
+        filing.comments = flattenAndSortComments(comments)
+      } catch (error) {
+      // set property to null to retry next time
+        filing.comments = null
+        // eslint-disable-next-line no-console
+        console.log('loadComments() error =', error)
+      // FUTURE: enable some error dialog?
+      }
+      // update comments count
+      filing.commentsCount = filing.comments?.length || 0
+
+      /** Flattens and sorts an array of comments. */
+      function flattenAndSortComments (comments: Array<CommentIF>): Array<CommentIF> {
+        if (comments && comments.length > 0) {
+        // first use map to change comment.comment to comment
+          const temp: Array<any> = comments.map(c => c.comment)
+          // then sort newest to oldest
+          // NB: these `new Date()` are safe because we're comparing like units
+          temp.sort((a, b) => new Date(a.timestamp) < new Date(b.timestamp) ? 1 : -1)
+          return temp
+        }
+        return []
+      }
+    },
+
+    /** Loads the documents for this history item. */
+    async loadDocuments (filing: ApiFilingIF): Promise<void> {
+      const businessStore = useBusinessStore()
+      const rootStore = useRootStore()
+      try {
+      // fetch documents object from API
+        const documents = await LegalServices.fetchDocuments(filing.documentsLink)
+        // load each type of document
+        filing.documents = []
+        // iterate over documents properties
+        for (const prop in documents) {
+          if (prop === 'legalFilings' && Array.isArray(documents.legalFilings)) {
+          // iterate over legalFilings array
+            for (const legalFiling of documents.legalFilings) {
+            // iterate over legalFiling properties
+              for (const prop in legalFiling) {
+              // this is a legal filing output
+                let title: string
+                // use display name for primary document's title
+                if (prop === filing.name) title = filing.displayName
+                else title = EnumUtilities.filingTypeToName(prop as FilingTypes, null)
+
+                const date = DateUtilities.dateToYyyyMmDd(new Date(filing.submittedDate))
+                const filename = `${businessStore.getIdentifier} ${title} - ${date}.pdf`
```

Comment:
Instead of getting the identifier here each time through this for loop, I suggest using a local variable outside the loop.

Ditto for staff role on line 268.

Reply:
Eg,
`const identifier = useBusinessStore().getIdentifier`

Reply:
Great, thanks.